### PR TITLE
store: Update pools's close function

### PR DESCRIPTION
**Pull request by zimulala**
The pools can close when there are some connections not put back pools.

### CODE REVIEW

**Comment by zimulala** [comment]
PTAL @coocood @disksing @tiancaiamao

**Comment by coocood** [comment]
LGTM

File: store/tikv/conn_pool_test.go [hunk 1]
```diff
@@ -93,3 +93,39 @@ func (s *testPoolSuite) TestPool(c *C) {
...
+		pools.PutConn(conns[i])
+	}
+
+	ch := make(chan struct{})
+	go func() {
+		<-ch
+		pools.PutConn(conns[capability-1])
+	}()
+	close(ch)
+	pools.Close()
```
Comment (tiancaiamao) [inline store/tikv/conn_pool_test.go]: It's a potential resource leak. 
`Close()` close just the conns  at that moment, but the conn put back later will never be closed.

**Comment by tiancaiamao** [comment]
ok, that's not a big problem, just in test code.
LGTM

**Comment by disksing** [comment]
LGTM.

**Comment by tiancaiamao** [comment]
LGTM

**Pull request merged**